lib/arborist/build-ideal-tree.js

```diff
@@ -464,13 +469,38 @@ module.exports = cls => class IdealTreeBuilder extends Tracker(Virtual(Actual(cl
     // satisfied by whatever's in that file anyway.
     if (this[_depsSeen].has(node) ||
         node.root !== this.idealTree ||
-        node.hasShrinkwrap)
+        node.hasShrinkwrap && !this[_complete])
       return this[_buildDepStep]()
 
     this[_depsSeen].add(node)
     this[_currentDep] = node
     process.emit('time', `idealTree:${node.location || '#root'}`)
 
+    // if we're loading a _complete_ ideal tree, for a --package-lock-only
+    // installation for example, we have to crack open the tarball and
+    // look inside if it has bundle deps or shrinkwraps.  note that this is
+    // not necessary during a reification, because we just update the
+    // ideal tree by reading bundles/shrinkwraps in place.
+    if (this[_complete] && node !== this.idealTree) {
+      const Arborist = this.constructor
+      const bd = node.package.bundleDependencies
+      const hasBundle = bd && Array.isArray(bd) && bd.length
+      const { hasShrinkwrap } = node
+      if (hasBundle || hasShrinkwrap) {
+        const opt = { ...this.options }
+        await cacache.tmp.withTmp(this.cache, opt, async path => {
+          await pacote.extract(node.resolved, path, opt)
+
+          if (hasShrinkwrap)
+            await new Arborist({ ...this.options, path })
+              .loadVirtual({ root: node })
```

Comment:
wow are these composing install trees? 🤩 is it that easy? it gives me some workspaces ideas...

Reply:
Yep!  https://github.com/npm/arborist/issues/72 is an open issue to add a similar facility to `loadActual()`, which will make the loading of bundle deps a bit easier.